paymentsheet/src/test/java/com/stripe/android/paymentsheet/flowcontroller/DefaultPaymentSheetLoaderTest.kt

```diff
@@ -84,102 +83,106 @@ internal class DefaultFlowControllerInitializerTest {
-                        isGooglePayReady = false
-                    )
+    fun `init without configuration should return expected result`() = runTest {
+        val expectedIntent = PaymentIntentFixtures.PI_REQUIRES_PAYMENT_METHOD
+
+        val initializer = createInitializer()
+
+        assertThat(
+            initializer.load(
+                clientSecret = PaymentSheetFixtures.PAYMENT_INTENT_CLIENT_SECRET,
+                paymentSheetConfiguration = null,
+            )
+        ).isEqualTo(
+            PaymentSheetLoader.Result.Success(
+                PaymentSheetState.Full(
+                    config = null,
+                    clientSecret = PaymentSheetFixtures.PAYMENT_INTENT_CLIENT_SECRET,
+                    stripeIntent = expectedIntent,
+                    customerPaymentMethods = emptyList(),
+                    savedSelection = SavedSelection.Link,
+                    isGooglePayReady = false,
                 )
             )
-        }
+        )
+    }
 
     @Test
     fun `init with configuration should return expected result`() = runTest {
         prefsRepository.savePaymentSelection(
             PaymentSelection.Saved(PaymentMethodFixtures.CARD_PAYMENT_METHOD)
         )
 
+        val expectedIntent = PaymentIntentFixtures.PI_REQUIRES_PAYMENT_METHOD
+
+        val initializer = createInitializer()
+
         assertThat(
-            initializer.init(
+            initializer.load(
                 PaymentSheetFixtures.PAYMENT_INTENT_CLIENT_SECRET,
                 PaymentSheetFixtures.CONFIG_CUSTOMER_WITH_GOOGLEPAY
             )
         ).isEqualTo(
-            FlowControllerInitializer.InitResult.Success(
-                InitData(
-                    PaymentSheetFixtures.CONFIG_CUSTOMER_WITH_GOOGLEPAY,
-                    PaymentSheetFixtures.PAYMENT_INTENT_CLIENT_SECRET,
-                    PaymentIntentFixtures.PI_REQUIRES_PAYMENT_METHOD,
-                    PAYMENT_METHODS,
-                    SavedSelection.PaymentMethod(
-                        id = "pm_123456789"
-                    ),
-                    isGooglePayReady = true
+            PaymentSheetLoader.Result.Success(
+                PaymentSheetState.Full(
+                    config = PaymentSheetFixtures.CONFIG_CUSTOMER_WITH_GOOGLEPAY,
+                    clientSecret = PaymentSheetFixtures.PAYMENT_INTENT_CLIENT_SECRET,
+                    stripeIntent = expectedIntent,
+                    customerPaymentMethods = PAYMENT_METHODS,
+                    savedSelection = SavedSelection.PaymentMethod(id = "pm_123456789"),
+                    isGooglePayReady = true,
                 )
             )
         )
     }
 
     @Test
-    fun `init() with no customer, and google pay ready should default saved selection to google pay`() =
-        runTest {
-            prefsRepository.savePaymentSelection(null)
+    fun `init() with no customer and google pay ready should default to Link`() = runTest {
```

Comment:
We now expect Link instead of Google Pay because we no longer inject `isLinkEnabled = false`. With Link enabled, it’s supposed to take precedence over Google Pay.